src/app/src/main/kotlin/org/imperial/mrc/hint/clients/FuelClient.kt

```diff
@@ -87,4 +93,30 @@ abstract class FuelClient(protected val baseUrl: String)
                 .second
                 .asResponseEntity()
     }
+
+    private fun getRequest(uri: URI): HttpRequest
+    {
+        val entry = standardHeaders().entries.first()
+
+        return HttpRequest
+            .newBuilder()
+            .uri(uri)
+            .header(entry.key, entry.value.toString())
+            .GET()
+            .build()
+    }
+
+    private fun clientBuilder(): Builder
+    {
+        return HttpClient
+            .newBuilder()
+            .followRedirects(HttpClient.Redirect.ALWAYS)
+    }
+
+    protected fun getFile(uri: URI): HttpResponse<InputStream>
```

Comment:
Is this always a file? Looks like this function is more generic at this level?

Reply:
It's pulling a stream, so I guess it could be `getStream`..?
So as far as I can tell, this is replacing the old `java.net` implementation with the `java.net.http` equivalent, and moving it from the ADRFuelClient to FuelClient class to make it more generic. Makes sense!